www/docs/js/qual2.js

```diff
@@ -0,0 +1,165 @@
+var researchUser;
+var startTime = new Date();
+
+$(document).ready(function() {
+
+  // Retrieve the bucket from storage, or allocate the user if not.
+  if ($.localStorage.isEmpty('research.qual2')) {
+    researchUser = {
+      'bucket': Math.floor(Math.random() * 3) + 1,
+    };
+    $.localStorage.set('research.qual2', researchUser);
+  } else {
+    researchUser = $.localStorage.get('research.qual2');
+  }
+
+  // Record this page view and bucket
+  $.ajax({
+    url: '/action/record.php',
+    type: 'POST',
+    data: {
+      'method': 'view',
+      'page': document.URL,
+      'bucket': researchUser.bucket
+    },
+    dataType: 'json'
+  });
+
+  // Choose which action to take
+  if (researchUser.bucket == 1) {
+
+    // Bind the "where next" click event
+    $('#research-qual2-bucket1-wherenext').click(function(e){
+
+      e.preventDefault();
+
+      var link = $(this);
+
+      var endTime = new Date();
+      var timer = endTime - startTime;
+      timer /= 1000;
+
+      $.ajax({
+        url: '/action/record.php',
+        type: 'POST',
+        data: {
+          'method': 'click_popup_link',
+          'page': document.URL,
+          'bucket': researchUser.bucket,
+          'data': 'where-next',
+          'timer': timer,
+          'timeout': 300
+        },
+        dataType: 'json'
+      }).always(function() {
+        window.location.assign(link.attr('href'));
+      });
+
+    });
+
+    // Bind the "where next" click event
+    $('[data-research-qual2-bucket1-linkname]').click(function(e){
+
+      e.preventDefault();
+
+      var link = $(this);
+
+      var endTime = new Date();
+      var timer = endTime - startTime;
+      timer /= 1000;
+
+      $.ajax({
+        url: '/action/record.php',
+        type: 'POST',
+        data: {
+          'method': 'click_popup_link',
+          'page': document.URL,
+          'bucket': researchUser.bucket,
+          'data': link.data('research-qual2-bucket1-linkname'),
+          'timer': timer,
+          'timeout': 300
+        },
+        dataType: 'json'
+      });
+
+      $('#whereNextModal').foundation('reveal', 'close');
+
+    });
```

Comment:
Possible, but they also have different timeout behaviour and differ in how they execute the next action, so I'm inclined to say it's actually clearer to leave them as two separate bits of code since they conceptually do two different things (one reports back and closes the dialogue, one blocks the page being changed until the report is complete or has timed out).